dali/pipeline/executor/executor2/exec2.cc

```diff
@@ -158,20 +159,27 @@ class Executor2::Impl {
     if (output_order.has_value() && output_order != ws.output_order()) {
       // Set the order of the outputs to the requested output_order - no synchronization
       // is necessary, the stream has been properly synchronized a few lines above.
+      std::unordered_set<void *> already_processed(ws.NumOutput());
       for (int i = 0; i < ws.NumOutput(); i++) {
         if (ws.OutputIsType<GPUBackend>(i)) {
-          auto &output = ws.Output<GPUBackend>(i);
-          assert(output.ready_event() == pipe_out.event.get());
```

Comment:
This was raised when the same buffer was returned many times. See line 168 - we clear the event, so a subsequent encounter of the same buffer resulted in a mismatch.

Reply:
Not ops - pipelines.